clientconn.go

```diff
@@ -129,6 +132,13 @@ func WithBalancer(b Balancer) DialOption {
 	}
 }
 
+// WithServiceConfig returns a DialOption which has a channel to read the service configuration.
+func WithServiceConfig(c <-chan ServiceConfig) DialOption {
```

Comment:
To be clear, I'm not talking about interprocess communication here.  This is really an API issue, and it has nothing to do with how the code is spread across processes.

The desired goal here is that when a team uses multiple languages, they will want to write resolvers and tests in each language, and they should be able to encode the service config in the same form in all languages.  I think that means that we need to use JSON in the primary API that will be used in resolvers and in tests to pass the service config through.  And we should use this same form for the API that returns the service config to the application from the client channel.

If there's a way to meet this goal without picking up a JSON dependency in the grpc package, then I'm fine with that.  But that basically requires that whatever external package has the JSON support is always available for teams that use the service config, and that users never directly use the underlying APIs from the grpc package.

If you'd like to discuss this further, please throw something on my calendar.  Thanks!